upgradetool: JsonIOException and never finishing

Using snapshot 3466, there is progress, the tool now does something. It even handles some of my items and then there is an exception and the tool never finished (I killed it with Ctrl+C):
```
[main] INFO org.openhab.core.tools.internal.Upgrader - Copying item unit from state description to metadata in database '.\jsondb\org.openhab.core.items.Item.json'
[main] INFO org.openhab.core.tools.internal.Upgrader - uom3: Wrote 'unit=mm' to metadata.
[main] WARN org.openhab.core.tools.internal.Upgrader - uom4: State description contains unit place-holder '%unit%', check if 'unit' metadata is needed!
Exception in thread "main" com.google.gson.JsonIOException: Failed making field 'java.time.ZonedDateTime#dateTime' accessible; either change its visibility or write a custom TypeAdapter for its declaring type
        at com.google.gson.internal.reflect.ReflectionHelper.makeAccessible(ReflectionHelper.java:23)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:203)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:112)
        at com.google.gson.Gson.getAdapter(Gson.java:531)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.createBoundField(ReflectiveTypeAdapterFactory.java:137)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:211)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:112)
        at com.google.gson.Gson.getAdapter(Gson.java:531)
        at com.google.gson.Gson.toJson(Gson.java:778)
        at com.google.gson.Gson.toJsonTree(Gson.java:670)
        at com.google.gson.Gson.toJsonTree(Gson.java:649)
        at org.openhab.core.storage.json.internal.JsonStorage.put(JsonStorage.java:158)
        at org.openhab.core.tools.internal.Upgrader.itemCopyUnitToMetadata(Upgrader.java:124)
        at org.openhab.core.tools.UpgradeTool.main(UpgradeTool.java:79)
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make field private final java.time.LocalDateTime java.time.ZonedDateTime.dateTime accessible: module java.base does not "opens java.time" to unnamed module @530a271
        at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
        at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
        at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:178)
        at java.base/java.lang.reflect.Field.setAccessible(Field.java:172)
        at com.google.gson.internal.reflect.ReflectionHelper.makeAccessible(ReflectionHelper.java:20)
        ... 13 more
```
@J-N-K : any idea ?

--- a/tools/upgradetool/src/main/java/org/openhab/core/tools/internal/Upgrader.java
+++ b/tools/upgradetool/src/main/java/org/openhab/core/tools/internal/Upgrader.java
@@ -78,7 +78,8 @@ public class Upgrader {
             logger.error("Cannot access item database '{}', check path and access rights.", itemJsonDatabasePath);
             return;
         }
-        if (!Files.isWritable(metadataJsonDatabasePath)) {
+        // missing metadata database is also fine, we create one in that case
+        if (!Files.isWritable(metadataJsonDatabasePath) && Files.exists(metadataJsonDatabasePath)) {
             logger.error("Cannot access metadata database '{}', check path and access rights.",
                     metadataJsonDatabasePath);
             return;
@@ -164,10 +165,10 @@ public class Upgrader {
     }
 
     private static class UpgradeRecord {
-        public final ZonedDateTime executionDate;
+        public final String executionDate;
 
         public UpgradeRecord(ZonedDateTime executionDate) {
-            this.executionDate = executionDate;
+            this.executionDate = executionDate.toString();
         }
     }
 }

